server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepository.java

```diff
@@ -85,22 +85,22 @@ public MailKey store(Mail mail) throws MessagingException {
 
     @Override
     public Mail retrieve(MailKey key) {
-        return CompletableFutureUtil
-            .unwrap(mailDAO.read(url, key)
-                .thenApply(optional -> optional.map(this::toMail)))
-            .join()
-            .orElse(null);
+        return mailDAO.read(url, key)
+            .map(optional -> optional.map(this::toMail))
+            .flatMap(Mono::justOrEmpty)
+            .flatMap(Function.identity())
+            .defaultIfEmpty(null)
```

Comment:
I do not think it will works, try:

```
.blockOptional()
.orElse(null);
```

Reply:
I think `.flatMap(Function.identity())` is redundant

Reply:
> I think .flatMap(Function.identity()) is redundant

It's not, you need to flatten twice and there's no flatten method on flux